src/TADA_DataRetrieval_new.R

```diff
@@ -0,0 +1,369 @@
+#' dataframe. Note that the inputs (e.g. project, organization, siteType) with the
+#' exceptions of endDate and startDate match the web service call format from the
+#' online WQP GUI. endDate and startDate match the format suggested in USGS's
+#' dataRetrieval package (endDate = "YYYY-MM-DD"), which is a more familiar date
+#' format for R users than the WQP GUI's endDateHi = "MM-DD-YYYY".
+#'
+#' Multiple fields are queried together using AND logic, but multiple values within
+#' one field are queried together using OR logic. For example, within
+#' characteristicName, if you enter, c("pH", "Dissolved oxygen (DO)), the
+#' function will return all results that are "pH" OR "Dissolved oxygen (DO)". Similarly,
+#' if you enter c("VA", "IL"), the function will return results from Virginia OR Illinois.
+#' But the combo of these fields are ANDs: The function will return any pH and DO data
+#' from only Virginia or Illinois; the data must fit into one of the values from BOTH
+#' of the query fields.
+#' characteristicName and Characteristic Group also work as an AND, therefore the
+#' characteristicName must fall within the Characteristic Group when both are entered.
+#' sf, huc, countycode, and state also work as ANDs, therefore the huc, countycode, and 
+#' state must fall within the sf object when any combination of these are entered.
+#'
+#'
+#' Users can reference the \href{https://www.epa.gov/waterdata/storage-and-retrieval-and-water-quality-exchange-domain-services-and-downloads}{WQX domain tables}
+#' to find allowable values for queries, e.g., reference the WQX domain table to find countycode and statecode: https://cdx.epa.gov/wqx/download/DomainValues/County_CSV.zip
+#' Alternatively, you can use the WQP services to find areas where data is available in the US: https://www.waterqualitydata.us/Codes/countycode
+#'
+#' TADA_DataRetrieval automatically runs TADA_AutoClean on the incoming dataset. TADA_AutoClean
+#' is important for categorizing result value and detection limit data, as well as
+#' harmonizing key columns used in TADA. See ?TADA_AutoClean for more `````````````````````````````information.
+#'
+#' Note: TADA_DataRetrieval (by leveraging dataRetrieval),  automatically converts
+#' the date times to UTC. It also automatically converts the data to dates,
+#' datetimes, numerics based on a standard algorithm. See: ?dataRetrieval::readWQPdata
+#'
+#' @param startDate Start Date string in the format YYYY-MM-DD, for example, "2020-01-01"
+#' @param endDate End Date string in the format YYYY-MM-DD, for example, "2020-01-01"
+#' @param countycode FIPS county name. Note that a state code must also be supplied (e.g. statecode = "AL", countycode = "Chilton").
+#' @param huc A numeric code denoting a hydrologic unit. Example: "04030202". Different size hucs can be entered.
+#' @param siteid Unique monitoring location identifier
+#' @param sf A polygon to subset the WQP data by
```

Comment:
I wonder if an argument name like `shape` or `aoi` might be clearer than `sf`?

Reply:
i like that!